alphashield/database/mongodb_client.py

```
@@ -1,132 +1,73 @@
+    def store_agent_decision(self, decision: Dict[str, Any]) -> None:
+        # Validate via Pydantic, augment timestamp if missing
+        try:
+            if "timestamp" not in decision:
+                decision["timestamp"] = datetime.utcnow()
+            DecisionDoc(**decision)  # validation
+            # idempotency: unique (agent_id, loan_id, timestamp minute) simplistic
+            key = (decision["agent_id"], decision["loan_id"], str(decision["timestamp"])[:16])
+            if any((d.get("agent_id"), d.get("loan_id"), str(d.get("timestamp"))[:16]) == key for d in self.decisions):
+                return
+            self.decisions.append(dict(decision))
+        except Exception as e:
```

Ruff (0.14.1)

Do not catch blind exception: `Exception`

(BLE001)